### PR TITLE
The splash screen with navigation component was implemented

### PR DESCRIPTION
**Pull request by ezequielsetzesglobant**
Implementation of the application's splash screen with navigation component was carried out.

Pixel
https://user-images.githubusercontent.com/84338513/135319729-b280015e-124b-4850-88b6-341fe43c9fd0.mp4

Nexus
https://user-images.githubusercontent.com/84338513/135319752-10fb0731-233a-4a2c-a83a-15ce9c419688.mp4

Unit test
image

Coverage
image

### CODE REVIEW

File: app/src/main/java/com/example/pokemonapp/viewmodel/SplashScreenViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    companion object {
+        private const val WAIT_TIME: Long = 4000
+    }
+}
+
+data class SplashScreenData(var responseType: SplashScreenStatus, var data: Boolean = false, var error: String = ERROR_DEFAULT) {
+    companion object {
+        private const val ERROR_DEFAULT: String = "error default"
```
Comment (SartiMau) [inline app/src/main/java/com/example/pokemonapp/viewmodel/SplashScreenViewModel.kt]: Please put this in the companion of the SplashScreenViewModel.
The data class and the enum class should be inside the SplashScreenViewModel too

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/pokemonapp/fragment/SplashScreenFragment.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        binding = FragmentSplashScreenBinding.inflate(inflater, container, false)
+        viewModel.getLiveDataState().observe(viewLifecycleOwner, updateUIObserver)
+        viewModel.navigateToMainFragment()
+        return binding.root
+    }
+
+    private val updateUIObserver = Observer<Event<SplashScreenData>> { event ->
+        val eventContent = event.getContentIfNotHandled()
```
Comment (lucianoeli) [inline app/src/main/java/com/example/pokemonapp/fragment/SplashScreenFragment.kt]: you can avoid this variable, and just call it on when

File: app/src/main/java/com/example/pokemonapp/viewmodel/SplashScreenViewModel.kt (hunk 1 above)
Comment (lucianoeli) [reply app/src/main/java/com/example/pokemonapp/viewmodel/SplashScreenViewModel.kt:34]: you have no service, so i cant imagine a error here. actually you never use the error, nor the data... so i think you should only use the SplashScreenStatus

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1,3 +1,5 @@
 <resources>
     <string name="app_name">PokemonApp</string>
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (lucianoeli) [inline app/src/main/res/values/strings.xml]: delete this one

File: app/src/main/java/com/example/pokemonapp/viewmodel/SplashScreenViewModel.kt (hunk 1 above)
Comment (lucianoeli) [reply app/src/main/java/com/example/pokemonapp/viewmodel/SplashScreenViewModel.kt]: you re not using neither the data nor the error. cause you dont call any service here.
i think you should delete this and only use the enum class

Comment (SartiMau) [reply app/src/main/java/com/example/pokemonapp/viewmodel/SplashScreenViewModel.kt]: @ezequielsetzesglobant Remember that we do not put things that we are not using now

File: app/src/main/java/com/example/pokemonapp/fragment/SplashScreenFragment.kt (hunk 2 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/pokemonapp/fragment/SplashScreenFragment.kt]: done

File: app/src/main/java/com/example/pokemonapp/viewmodel/SplashScreenViewModel.kt (hunk 1 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/pokemonapp/viewmodel/SplashScreenViewModel.kt]: done

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (ezequielsetzesglobant) [reply app/src/main/res/values/strings.xml]: done

**Review by lucianoeli — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Pull request merged**